server/graphql/v2/object/Amount.js

```diff
@@ -25,7 +26,11 @@ export const Amount = new GraphQLObjectType({
     valueInCents: {
       type: GraphQLInt,
       resolve(amount) {
-        return parseInt(amount.value, 10);
+        if (isNil(amount.valueInCents)) {
```

Comment:
The object is never built from `valueInCents`, we always build it by passing a `value` 

```suggestion
        if (isNil(amount.value)) {
```